src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.ReinferredVariableTracker.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class NullableWalker
+    {
+#nullable enable
+        internal sealed class ReinferredVariableTracker
+        {
+            internal readonly struct Readonly
+            {
+                private readonly ReinferredVariableTracker? _tracker;
+                internal Readonly(ReinferredVariableTracker tracker)
+                {
+                    _tracker = tracker;
+                }
+
+                internal readonly T? GetReinferredSymbol<T>(T originalSymbol) where T : Symbol => _tracker?.GetReinferredSymbol(originalSymbol);
+                internal readonly ReinferredVariableTracker EnterNewFunction()
+                {
+                    Debug.Assert(_tracker is object);
+                    return _tracker?.EnterNewFunction()!;
+                }
+            }
+
+            private readonly Dictionary<Symbol, Symbol> _reinferredVariables;
```

Comment:
>_reinferredVariables [](start = 56, length = 20)

Are they all local variables? If not, perhaps use "Symbol" term instead. The same goes for the type names.

Reply:
There are parameters in here, I'll update the name.

---
In reply to: [327850790](https://github.com/dotnet/roslyn/pull/38769#discussion_r327850790) [](ancestors = 327850790)

Reply:
> There are parameters in here

What about local functions and lambdas?

---
In reply to: [327851162](https://github.com/dotnet/roslyn/pull/38769#discussion_r327851162) [](ancestors = 327851162,327850790)